### PR TITLE
fix: make code in build directory runnable

### PR DESCRIPTION
**Pull request by blakebyrnes**
This PR makes a few changes:
- build directory is runnable:
 - copies in the mitm/dist and replay/dist so it can be run after
 - you have to manually run yarn inside build once
- Removed shared/logger - moved to commons/Logger
- Create emulator-plugins-shared project to access the shared code instead of navigating backwards

### CODE REVIEW

**Comment by blakebyrnes** [comment]
Additional modifications:
- tests run build directory
- replay now builds typescript into build directory and works with those files
- copyfiles is used to copy non-typescript assets into the build project

File: core/lib/DomEnv.ts [hunk 1]
```diff
@@ -154,10 +154,10 @@ const typesonRegistryScript = fs.readFileSync(
 );
 
 const domStorageScript = fs.readFileSync(
-  require.resolve(`@secret-agent/injected-scripts/dist/scripts/domStorage.js`),
+  require.resolve(`@secret-agent/injected-scripts/scripts/domStorage.js`),
```
Comment (calebjclark) [inline core/lib/DomEnv.ts:157]: Does this show errors in WebStorm since it doesn't know about js build files from within Typescript files?

File: injected-scripts/tsconfig.json [hunk 2]
```diff
@@ -5,10 +5,12 @@
...
     "rootDir": ".",
-    "outDir": "./dist",
+    "outDir": "../build/injected-scripts",
     "composite": true,
+    "resolveJsonModule": true,
+    "moduleResolution": "node",
     "removeComments": true
   },
-  "include": ["interfaces/*.ts", "scripts/*.ts"],
+  "include": ["./interfaces/*.ts", "./scripts/*.ts", "./package.json"],
```
Comment (calebjclark) [inline injected-scripts/tsconfig.json:14]: That's cool. I didn't know non-TS files could be included from tsconfig.json!

File: replay/tsconfig.json [hunk 3]
```diff
@@ -1,9 +1,10 @@
 {
   "compilerOptions": {
-    "outDir": "dist",
+    "outDir": "../build/replay",
     "module": "esnext",
     "target": "es2018",
     "moduleResolution": "node",
+    "composite": true,
```
Comment (calebjclark) [inline replay/tsconfig.json:7]: Why is this needed?

Comment (calebjclark) [reply replay/tsconfig.json:7]: I read up on `composite` in the TS documentation, but I"m not quite sure why we need it here.

Comment (blakebyrnes) [reply replay/tsconfig.json:7]: Using the project references requires it

File: core/lib/DomEnv.ts (hunk 1 above)
Comment (blakebyrnes) [reply core/lib/DomEnv.ts:157]: It doesn't right now
